Src/Sarif.PatternMatcher.Sdk/ValidatorBase.cs

```diff
@@ -118,18 +118,75 @@ public static bool IsBase64EncodingOfPrintableCharacters(string secret)
             return true;
         }
 
-        public static bool ContainsDigitAndChar(string matchedPattern)
+        public static bool ContainsDigitAndChar(string secret)
         {
-            bool oneDigit = false, oneLetter = false;
+            return ContainAtLeastNDigit(secret, 1) && ContainAtLeastNLetter(secret, 1);
+        }
+
+        public static bool ContainLowerAndUpperCaseLetter(string secret)
+        {
+            return ContainAtLeastNLowerCaseLetter(secret, 1) && ContainAtLeastNUpperCaseLetter(secret, 1);
+        }
+
+        public static bool ContainAtLeastNDigit(string secret, int n)
```

Comment:
>ontainAtLeastNDigit [](http://example.com/codeflow?start=28&length=19)

Do we have any rules that look for more than one uppercase or lowercase letter or digit? I'm suspicious that this logic is unnecessary, and we'd be calling it with parameter of 1 at all times.

Reply:
VAULT_SERVICE_TOKEN seems to require this (shown at least two places). I will send the screenshot separately...